test/common/utils/wait_for.js

```diff
@@ -76,8 +80,8 @@ function blockchainReady(cb, retries, timeout, baseUrl) {
 	})();
 }
 
-function nodeStatus(cb) {
-	var request = popsicle.get(`${__testContext.baseUrl}/api/node/status`);
+function nodeStatus(cb, baseUrl) {
```

Comment:
We already done this mistake previously. I would be nice to not repeat it here. The `cb` should be last argument of the function as the best practices for NodeJS.

Reply:
I agree that having the callback as the last argument is the ideal approach. It was done like this to maintain backwards-compatibility. Note that we are already using this incorrect pattern for the `blockchainReady` function so at least it's consistent with the rest of the `wait_for` module.

The `wait_for` module is used everywhere in all our tests (functional, integration, ...); refactoring the `wait_for` module and all its usages would be out of scope for this issue. We should raise a separate issue for it.